pkg/sql/plan2/base_binder.go

```diff
@@ -994,6 +1000,13 @@ func (b *baseBinder) bindNumVal(astExpr *tree.NumVal) (*Expr, error) {
 	case tree.P_decimal:
 		return returnDecimalExpr(astExpr.String())
 	case tree.P_float64:
+		originString := astExpr.String()
+		if !strings.Contains(originString, "e") {
```

Comment:
ContainsAny "eE"?

Reply:
parser had changed originString to lowcase